playerbot/strategy/values/ItemCountValue.cpp

```diff
@@ -40,6 +40,40 @@ list<Item*> InventoryItemValue::Calculate()
     return Find(ai, qualifier);
 }
 
+list<Item*> EquipedUsableRingValue::Calculate()
+{
+   list<Item*> rings;
+   list<Item*> result;
+
+   Player* bot = ai->GetBot();
+
+   if (Item* ring1 = bot->GetItemByPos(INVENTORY_SLOT_BAG_0, EQUIPMENT_SLOT_FINGER1))
+      rings.push_back(ring1);
+
+   if (Item* ring2 = bot->GetItemByPos(INVENTORY_SLOT_BAG_0, EQUIPMENT_SLOT_FINGER2))
+      rings.push_back(ring2);
+
+   if (rings.empty())
+      return result;
```

Comment:
I think the list<Item*> rings is not needed, you should put the spell check and cooldown condition in a separate method and pass the ring there, if the method returns true then the ring is valid